packages/firebase_messaging/firebase_messaging/android/src/main/java/io/flutter/plugins/firebase/messaging/FlutterFirebaseMessagingPlugin.java

```diff
@@ -613,4 +614,10 @@ public Task<Void> didReinitializeFirebaseCore() {
 
     return taskCompletionSource.getTask();
   }
+
+  private Map<String, Object> safelyCastToMap(Object obj) {
+    @SuppressWarnings("unchecked")
+    Map<String, Object> result = (Map<String, Object>) obj;
+    return result;
+  }
```

Comment:
Is this here just to get rid of the warning? if so shouldn't it be renamed to `uncheckedCastToMap`?

Reply:
Yeah, unfortunately, SuppressWarnings are only allowed in certain contexts. I had to take that cast out in order to suppress it. I only suppress warnings on API that I believe to be stable (i.e. no occurrences of failure).

Reply:
but we need to change it to `uncheckedCastToMap` for example, to give the intention that we know that's safe to cast it unchecked. what do you think?